ansible/playbooks/do_setup.yml

```diff
@@ -200,20 +203,29 @@
 
   # Signing + DNS
   # ===========================================
-    - name: d_coean | signing | set signing domain + authtoken
+    - name: d_ocean | signing | set signing domain + authtoken
       ansible.builtin.set_fact:
         full_signing_domain: "{{ signing_host }}.{{ full_domain }}"
         signing_subdomain: "{{ signing_host + '.' + subdomain if subdomain else signing_host }}"
         signing_authtoken: "{{ 99999999 | random | to_uuid }}"
 
       when: enable_signing
+      changed_when: true
+
+    - name: d_ocean | signing | check signing record exists
+      ansible.builtin.command: doctl compute domain records ls {{ domain }} -o json
+      register: check_domain
+      changed_when: true
 
     - name: d_ocean | signing | register the dns for signing subdomain
       ansible.builtin.command: >-
           doctl compute domain records create --record-type A --record-name "{{ signing_subdomain }}" --record-data "{{ lb_ip }}" "{{ domain }}"
 
       register: signing_dns_result
-      when: enable_signing
+      when: "check_domain.stdout | from_json | json_query(name_query) | length < 1"
+      changed_when: true # if it ran, it changed
+      vars:
+        name_query: '[?name==`{{ signing_subdomain }}`]'
```

Comment:
makes signing record creation idempotent. A bit spicy for this otherwise large but low signal PR. Thought I'd point it out explicitly.